generator/cli/generate-migration-sql.ts

```diff
@@ -0,0 +1,106 @@
+import execa from 'execa'
+import * as FS from 'fs-jetpack'
+import { log } from 'console'
+import { getTemplateInfos } from './generate-type-script'
+import { tools } from '../../src/fileTransformer/fileTransformer'
+import { File } from '../../src/types'
+import { Data } from '~/src/data'
+import prependFile from 'prepend-file'
+export type DatasourceProvider = Exclude<Data.DatasourceProviderName, 'mongodb' | 'sqlite'>
+
+export default async function generateMigrationSql(params: {
+  templatesRepoDir: string
+  outputDir: string
+}): Promise<void> {
+  log(`generating migration sql for each template x provider to ${params.outputDir}`)
+
+  FS.remove(params.outputDir)
+
+  const templateInfos = getTemplateInfos({ templatesRepoDir: params.templatesRepoDir })
+
+  log(`Found templates:`, { templates: templateInfos.map((t) => t.displayName) })
+
+  const indexFile = `${params.outputDir}/index.ts`
+  FS.write(indexFile, '')
```

Comment:
This file depends on the generated migration modules.

Generate this once last with all the source code it will need.